tools/pipelines/templates/include-publish-docker-service-steps.yml

```diff
@@ -28,52 +28,48 @@ parameters:
   default:
 
 jobs:
-- deployment: publish_${{ replace(parameters.environment, '-', '_') }}
-  displayName: Publish ${{ parameters.environment }}
-  environment: ${{ parameters.environment }}
```

Comment:
Not so sure about this one or the usage in `include-publish-npm-package-*` ... we do have some approvals/checks in the Environment objects in ADO to make sure only certain branches can use them and thus publish to public npm, or the container registry repository that gets mirrored to docker hub, for example.

Reply:
Yeah, this seems worth preserving. I didn't notice it on first pass, but the 1es templates include an explicit release job for publishing rust crates to registries, which is analogous to the npm & docker cases we use today. So it probably makes sense to migrate to those instead, and if that isn't possible today (would be because we need support for oss case), we can hold off on doing so until we get support.